Content.Server/GameObjects/Components/Morgue/CrematoriumEntityStorageComponent.cs

```diff
@@ -63,15 +70,27 @@ public override bool CanOpen(IEntity user, bool silent = false)
             return base.CanOpen(user, silent);
         }
 
-        public void Cremate()
+        public void TryCremate()
         {
             if (Cooking) return;
             if (Open) return;
 
+            Cremate();
+        }
+
+        public void Cremate()
+        {
+            if (Open)
+                CloseStorage();
+
             Appearance?.SetData(CrematoriumVisuals.Burning, true);
             Cooking = true;
 
-            Timer.Spawn(_burnMilis, () =>
+            if (_cremateCancelToken != null)
+                _cremateCancelToken.Cancel();
+
+            _cremateCancelToken = new CancellationTokenSource();
+            Robust.Shared.Timers.Timer.Spawn(_burnMilis, () =>
             {
                 Appearance?.SetData(CrematoriumVisuals.Burning, false);
```

Comment:
Check if either owner is deleted first (you pretty much always need to when dealing with timers).

Reply:
What do you mean by either? So far I've added a check inside the timer for Owner.Deleted, are there other checks that I'm missing? Trying to search the codebase for examples but I can't find any

Reply:
Sorry when I started writing it I thought there was another entity being cooked.